packages/app-shell-shared/src/types/Model.ts

```diff
@@ -0,0 +1,86 @@
+import type { ServiceConfig } from "@hitachivantara/app-shell-services";
+
+import type { UseCondition } from "./condition";
+import type {
+  HvAppShellConditionConfig,
+  HvAppShellConfig,
+  HvAppShellHeader,
+  HvAppShellHeaderAction,
+  HvAppShellMainPanelConfig,
+  HvAppShellMenuConfig,
+  HvAppShellProvidersConfig,
+  HvAppShellTopViewConfig,
+  HvAppShellViewsConfig,
+} from "./Config";
+
+/**
+ * Processed condition with globalIndex being a unique number for each condition in the config and used in results lookup
+ */
+export type ProcessedCondition = HvAppShellConditionConfig & {
+  globalIndex: number;
+};
+
+type WithRequiredKey<T> = T & {
+  key: string;
+};
+
+export type ProcessedMenu = WithRequiredKey<
+  Omit<HvAppShellMenuConfig, "conditions" | "submenus">
+> & {
+  conditions?: ProcessedCondition[];
+  submenus?: ProcessedMenu[];
+};
+
+export type ProcessedView = WithRequiredKey<
+  Omit<HvAppShellViewsConfig, "conditions" | "views">
+> & {
+  conditions?: ProcessedCondition[];
+  views: ProcessedView[];
+};
+
+export type ProcessedTopView = ProcessedView &
+  Omit<HvAppShellTopViewConfig, keyof HvAppShellViewsConfig>;
+
+export type ProcessedHeaderAction = WithRequiredKey<
+  Omit<HvAppShellHeaderAction, "conditions">
+> & {
+  conditions?: ProcessedCondition[];
+};
+
+export type ProcessedProvider = WithRequiredKey<
+  Omit<HvAppShellProvidersConfig, "conditions">
+> & {
+  conditions?: ProcessedCondition[];
+};
+
+// Cannot use 'WithRequiredKey' and 'Omit' here due to ServiceConfig being a union type
+// from
+export type ProcessedServiceConfig = ServiceConfig & {
+  key: string;
+  conditions?: ProcessedCondition[];
+};
+
+export type ProcessedServices = Record<string, ProcessedServiceConfig[]>;
+
+export type ProcessedMainPanel = Omit<HvAppShellMainPanelConfig, "views"> & {
+  views?: ProcessedTopView[];
+};
+
+export type ProcessedHeader = Omit<HvAppShellHeader, "actions"> & {
+  actions: ProcessedHeaderAction[];
+};
+
+export type AppShellModel = Omit<
+  HvAppShellConfig,
+  "menu" | "mainPanel" | "header" | "providers" | "services"
+> & {
+  menu?: ProcessedMenu[];
+  mainPanel?: ProcessedMainPanel;
+  header?: ProcessedHeader;
+  providers?: ProcessedProvider[];
+  services?: ProcessedServices;
+  /** All conditions extracted from the config, indexed by globalIndex */
```

Comment:
`All conditions extracted from the config, ...` -> `All conditions present in the App Shell model, ...`